cmake/dump_module_states.cmake

```diff
@@ -0,0 +1,58 @@
+# ############################################################################
+# Copyright (c) 2020 One Identity
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+# ############################################################################
+
+# Lists actual module definition results
+# Behaves similar to the cmake -L option but filters out the relevant information and add some highlighting
+#
+function(dump_module_states)
+  get_cmake_property(_variableNames VARIABLES)
+  list(SORT _variableNames)
+
+  message(STATUS "---------------------------------")
+  message(STATUS "Switchable modules enabled state:")
```

Comment:
I think we can't be sure these are modules only.

I'd just mimic a similar summary we have in autotools.

```
---------------------------------
syslog-ng Open Source Edition 3.38.1.260.geaac09a configured
---------------------------------

ENABLE_* On/Off